### PR TITLE
Kubernetes pack update

### PR DESCRIPTION
**Pull request by ritawatson**
This PR refactors the Kubernetes pack to include the following and adds a doc that describes how to update a pack based on recent input from @kreeuwijk.

- An explanation of Spectro Cloud's update policies for the Kubernetes pack
- An explanation of the deprecation policy for minor Kubernetes packs. 
- Prerequisites for each supported version
- Usage
- Terraform

💻 Doc Previews:
[Pack doc](https://deploy-preview-1158--docs-spectrocloud.netlify.app/integrations/kubernetes)
[Update Packs](https://deploy-preview-1158--docs-spectrocloud.netlify.app/integrations/update-packs/update-packs)
🎫 [DOC-610](https://spectrocloud.atlassian.net/browse/DOC-610)

### CODE REVIEW

**Comment by spectro-prow** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/spectrocloud/librarium/pull/1158#" title="Author self-approved">ritawatson</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign 
You can assign the PR to them by writing `/assign ` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=spectrocloud%2Flibrarium).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/spectrocloud/librarium/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: content/docs/06-integrations/00-kubernetes.md
Comment (karl-cardenas-coding) [inline content/docs/06-integrations/00-kubernetes.md:19]:
```suggestion
The Kubernetes pack defines the default properties that we use to deploy Kubernetes clusters. Most of the Kubernetes hardening standards that the Center for Internet Security (CIS) recommends are enabled by default. The Kubernetes pack supports most [cloud and data center infrastructure providers](/clusters).
```
This is a good spot for us to use we. Whenever possible, let's use "we" otherwise it can feel like we are creating distance between ourselves and docs.

File: content/docs/06-integrations/00-kubernetes.md
Comment (karl-cardenas-coding) [inline content/docs/06-integrations/00-kubernetes.md:49]: This could be me, but it was not obvious what the X represented. I'm guessing it means "not available"? It will help if we explain the table a bit.  In the second column of the table, I wonder if we state "Support Enabled" could clarify the confusion. 

We could also use these icons ✅ and ❌ to improve readability.  If you want to use those icons, you literally can copy and paste them to the markdown 😄

File: content/docs/06-integrations/00-kubernetes.md
Comment (karl-cardenas-coding) [inline content/docs/06-integrations/00-kubernetes.md:59]: I would place all the parameters in code ticks.

File: content/docs/06-integrations/00-kubernetes.md
Comment (karl-cardenas-coding) [inline content/docs/06-integrations/00-kubernetes.md:4]: Let's give a more insightful description for search engine optimization purposes.

**Review by karl-cardenas-coding — CHANGES_REQUESTED** [review]
I like the changes. I just have some concerns about the table and its readability.

File: content/docs/06-integrations/00-kubernetes.md
Comment (ritawatson) [reply content/docs/06-integrations/00-kubernetes.md:49]: Discussed with Karl and implemented his proposed table.

File: content/docs/06-integrations/00-kubernetes.md
Comment (ritawatson) [reply content/docs/06-integrations/00-kubernetes.md:59]: fixed

File: content/docs/06-integrations/00-kubernetes.md
Comment (ritawatson) [reply content/docs/06-integrations/00-kubernetes.md]: changed to say "Kubernetes pack that deploys containers in Palette"

**Review by karl-cardenas-coding — APPROVED** [review]

**Comment by spectro-prow** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/spectrocloud/librarium/pull/1158#pullrequestreview-1351452990" title="Approved">karl-cardenas-coding</a>*, *<a href="https://github.com/spectrocloud/librarium/pull/1158#" title="Author self-approved">ritawatson</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign 
You can assign the PR to them by writing `/assign ` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=spectrocloud%2Flibrarium).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/spectrocloud/librarium/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: content/docs/06-integrations/00-kubernetes.md
Comment (vishwanaths) [inline content/docs/06-integrations/00-kubernetes.md:61]: calicoNetworkCIDR - this will be present only if calico is used as CNI.  You can say that - "This should match the corresponding networking layer property. Eg: calicoNetworkCIDR. Same applies to other k8s versions below as well.
"

File: content/docs/06-integrations/00-kubernetes.md
Comment (vishwanaths) [inline content/docs/06-integrations/00-kubernetes.md:31]: Should we explain somewhere what is meant by deprecate , disable and delete? This is not yet available on the UI as of now and in proposal stage.

**Review by vishwanaths — CHANGES_REQUESTED** [review]

File: content/docs/06-integrations/00-kubernetes.md
Comment (brathina-spectro) [inline content/docs/06-integrations/00-kubernetes.md:76]: There's a typo in here. Should be "nodes or pods"

File: content/docs/06-integrations/00-kubernetes.md
Comment (ritawatson) [reply content/docs/06-integrations/00-kubernetes.md:31]: @vishwanaths This information is now added and will be moved to a separate doc about deprecated packs.

File: content/docs/06-integrations/00-kubernetes.md
Comment (ritawatson) [reply content/docs/06-integrations/00-kubernetes.md:69]: Good catch @vishwanaths. Removed mention of calicoNetworkCIDR.

File: content/docs/06-integrations/00-kubernetes.md
Comment (ritawatson) [reply content/docs/06-integrations/00-kubernetes.md:70]: Good catch @brathina-spectro. Fixed typo throughout doc.

**Review by karl-cardenas-coding — CHANGES_REQUESTED** [review]
Changes related to the new file, update-packs.

File: content/docs/03-cluster-profiles/5-update-profiles.md
Comment (kreeuwijk) [inline content/docs/03-cluster-profiles/5-update-profiles.md:30]: You cannot --> You should not

File: content/docs/03-cluster-profiles/5-update-profiles.md
Comment (kreeuwijk) [inline content/docs/03-cluster-profiles/5-update-profiles.md:33]: Updating to a newer version --> Updating to a newer Kubernetes version

File: content/docs/03-cluster-profiles/5-update-profiles.md
Comment (kreeuwijk) [inline content/docs/03-cluster-profiles/5-update-profiles.md:115]: If there are lines you did not add that are highlighted, they may have been removed in the new version, and you should be copy them over. --> If there are lines you did not add that are red highlighted, they have been removed in the new version, and you should *not* copy them over.

File: content/docs/06-integrations/00-kubernetes.md
Comment (kreeuwijk) [inline content/docs/06-integrations/00-kubernetes.md:80]: It turns out that this flag **cannot** be set to `False` as that breaks most cluster deployments. @sadysnaat I thought this flag was going to made hidden?

File: content/docs/06-integrations/00-kubernetes.md
Comment (kreeuwijk) [inline content/docs/06-integrations/00-kubernetes.md:81]: We are adding `pack:serviceDomain` in the next update (3.3.x), we may as well add that to the documentation too.

| ``pack:serviceDomain`` | The DNS name for the service domain in the cluster. Default: `cluster.local`|

File: content/docs/06-integrations/00-kubernetes.md
Comment (kreeuwijk) [inline content/docs/06-integrations/00-kubernetes.md:123]: #servicedomain: "cluster.local"

**Review by kreeuwijk — CHANGES_REQUESTED** [review]
Some small corrections, will approve once processed

File: content/docs/03-cluster-profiles/5-update-profiles.md
Comment (ritawatson) [reply content/docs/03-cluster-profiles/5-update-profiles.md:115]: Added the revised sentence.

File: content/docs/03-cluster-profiles/5-update-profiles.md
Comment (ritawatson) [reply content/docs/03-cluster-profiles/5-update-profiles.md:33]: Added the revision. Also removed mention of Kubernetes in the sentence that introduces the bullet list:
"When updating to a new pack version, these rules apply:"

File: content/docs/06-integrations/00-kubernetes.md
Comment (ritawatson) [reply content/docs/06-integrations/00-kubernetes.md:81]: Added this parameter.

File: content/docs/06-integrations/00-kubernetes.md
Comment (brathina-spectro) [inline content/docs/06-integrations/00-kubernetes.md:40]: Should this be disabled ? Both this and the next paragraph says Delete

**Review by karl-cardenas-coding — APPROVED** [review]
@ritawatson  looks good. I made some minor fixes to the spacing that you asked me about. I also fixed the **Disable** missing as called out @brathina-spectro 👍🏻

**Comment by spectro-prow** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/spectrocloud/librarium/pull/1158#pullrequestreview-1361666611" title="Approved">karl-cardenas-coding</a>*, *<a href="https://github.com/spectrocloud/librarium/pull/1158#" title="Author self-approved">ritawatson</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign 
You can assign the PR to them by writing `/assign ` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=spectrocloud%2Flibrarium).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/spectrocloud/librarium/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: content/docs/03-cluster-profiles/5-update-profiles.md
Comment (ritawatson) [reply content/docs/03-cluster-profiles/5-update-profiles.md]: revised wording

File: content/docs/06-integrations/00-kubernetes.md
Comment (ritawatson) [reply content/docs/06-integrations/00-kubernetes.md]: This is from the [Decprecation Schedule](https://spectrocloud.atlassian.net/wiki/spaces/ENGINEERIN/pages/1863843883/Packs+Deprecation). Leaving it as is.

CleanShot 2023-03-28 at 11 16 22

**Pull request merged**
